### DISCUSSION TITLE
Dataclasses

### DISCUSSION THREAD

**Discussion by jimman2003** [pymedusa/Medusa#9029] [Ideas & Feature requests]
What is opinion of the community for dataclasses in the code base for e.g Series,Media to eliminate some boilerplate..
Are you ok with the requirement of Python 3.7?
I might help with some porting...

**Comment by p0psicles** [comment]
I have no idea what dataclasses are? 
And yes? Py3.6 is currently the minimum supported version

**Reply by p0psicles** [reply]
Okay quick read. It could be a replacement for namedtuple. But it's introduced in 3.7.

Having that out of the way. What feature are you really requesting here?

**Reply by jimman2003** [reply]
The introduction of dataclasses for e.g Series,Media to eliminate boilerplate...

**Reply by p0psicles** [reply]
With what purpose?

**Reply by OmgImAlexis** [reply]
They don’t look like the worse idea. It would mean no support at all for under 3.7 though. 

https://realpython.com/python-data-classes/

**Reply by p0psicles** [reply]
Yeah it'll definitely be usable. But it looks to be a slimmed down version of the Class. So I'm curious what this `feature request` is about. 

Would be the same if someone did a feature request for some of your js projects. And requested for js proxies? 

Btw medusa is already fully OO.

**Reply by jimman2003** [reply]
@p0psicles I think you got it all wrong... Please skim at least the [Python Docs](https://docs.python.org/3/library/dataclasses.html).My understanding is that dataclasses are like auto-getters and auto-setters in java IDEs but at the language level
P.S I have seen some of the repo's code and I have observed the coding patterns... I already knew that classes are being used

**Reply by p0psicles** [reply]
Yeah I see. It has its place. I could use it for the recommended shows in future. But I'm pretty satisfied with the classes we have for Series, Episode, etc

**Reply by jimman2003** [reply]
Understood..
As a side note I believe the dataclass module has an similar third party library for python 3.6
